src/Playwright/Transport/WebSocketTransport.cs

```diff
@@ -0,0 +1,180 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net.WebSockets;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Playwright.Transport
+{
+    internal class WebSocketTransport : IConnectionTransport, IDisposable
+    {
+        private const int DefaultBufferSize = 16000;  // Byte buffer size
+        private readonly ClientWebSocket _webSocket;
+        private readonly string _wsEndpoint;
+        private readonly CancellationTokenSource _readerCancellationSource;
+        private readonly BrowserTypeConnectOptions _options;
+        private readonly int _slowMo;
+        private readonly int _timeout;
+
+        internal WebSocketTransport(
+            string wsEndpoint = default,
+            BrowserTypeConnectOptions options = default)
+        {
+            _webSocket = new ClientWebSocket();
+            _wsEndpoint = wsEndpoint;
+            _options = options;
+            _slowMo = _options?.SlowMo != null ? (int)_options?.SlowMo : 0;
+            _timeout = _options?.Timeout != null ? (int)_options?.Timeout : 30000;
+            _readerCancellationSource = new CancellationTokenSource(_timeout);
```

Comment:
Bug: if I read it correctly, after 30 seconds of operation this web socket will die. That's not our intent with timeout.

Reply:
fixed